### PR TITLE
feat: implement editor toolbar component

### LINKED ISSUE #24: feat: Implement Toolbar component
## Task Description
- A clear and concise description of what the task is.

## Objectives

- [x] Objective 1: The Toolbar may be wrapped in a Toolbar Group, or it may be independently inside the Toolbar as a Toolbar Item rather than a Toolbar group.
- [x] Objective 2: Think and implement basic styling on how to set provider.
- [x] Objective 3: Write each storybook.

## Time Estimate

- Estimated Duration: 2 days

### PR DESCRIPTION
**Pull request by rosielsh**
## Description

This PR implements the Toolbar component for our Markdown editor. It supports both horizontal and vertical orientations, visibility toggling, and custom styling options.

### Related Issue
- closed #24 

## Changes Made

- Change 1: Designed and implemented the Toolbar component with detailed props:
  - `orientation`: Allows setting the toolbar layout to 'horizontal' or 'vertical'.
  - `visible`: Controls the visibility of the toolbar.
  - `onVisibleChange`: Callback for visibility changes, enabling dynamic toolbar display.
  - Extended `React.HTMLAttributes<HTMLDivElement>` to support all standard HTML div attributes.

- Change 2: Created Storybook stories to showcase different Toolbar configurations:
  - Default horizontal layout
  - Vertical orientation
  - Hidden toolbar
  - Custom styled toolbar
  - Dynamic visibility toggle

- Change 3: Separated style-related logic into a util function `mergeStyles`.

- Change 4: Implemented `useVisibilityChange` hook for managing toolbar visibility.

## How Has This Been Tested?

- Test 1: Ran all Storybook stories to verify different configurations of the Toolbar component.
- Test 2: Manually tested toolbar orientation switching and visibility toggling in the Storybook environment.
- Test 3: Verified that custom styles can be applied without conflicting with orientation-based styles.

## Screenshots (if appropriate):
This is the storybook url from which this address was distributed. ([Go to the deployed site](https://66c84d12643d93af60910c10-szempfzvus.chromatic.com/?path=/docs/mkeditor-editorcontent--docs))

https://github.com/user-attachments/assets/99e2aba9-7c90-4578-94ab-ed3194a9de83

### API spec
image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/editor-content/editor-content.stories.tsx
Comment (GHWooo) [inline src/components/editor-content/editor-content.stories.tsx:1]: How is this change related to the PR?

File: src/components/editor-toolbar/editor-toolbar.tsx [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+  onVisibleChange,
+  ...props
+}: ToolbarProps) => {
+  const orientationStyle: CSSProperties = {
+    display: 'flex',
+    flexDirection: orientation === 'horizontal' ? 'row' : 'column',
+    gap: '2px',
+  };
+
+  const { style, otherProps } = mergeStyles<React.HTMLAttributes<HTMLDivElement>>(orientationStyle, props);
```
Comment (GHWooo) [inline src/components/editor-toolbar/editor-toolbar.tsx]: So, going forward, will you be managing styles within the component itself, like with `orientationStyle`?

File: src/components/editor-content/editor-content.stories.tsx
Comment (rosielsh) [reply src/components/editor-content/editor-content.stories.tsx:1]: It's not related to the PR, but it's a single file unit, so I put it in the PR. 😅
Will it be a problem?

File: src/components/editor-toolbar/editor-toolbar.tsx (hunk 1 above)
Comment (rosielsh) [reply src/components/editor-toolbar/editor-toolbar.tsx]: I think that's probably the case.
I thought it would be intuitive to write the parts that handle styles directly from the components, so I implemented them in this way.

File: src/components/editor-content/editor-content.stories.tsx
Comment (GHWooo) [reply src/components/editor-content/editor-content.stories.tsx:1]: It's not a big issue, but it's not the best practice either. 😅 Even if it's just a single line of code, if it's unrelated, it's better to create a separate PR. For minor changes like that, you can merge them directly.

Comment (rosielsh) [reply src/components/editor-content/editor-content.stories.tsx:1]: Oh,,, okay.
In the future, I will create additional PR. Thanks !

File: src/components/editor-toolbar/editor-toolbar.tsx (hunk 1 above)
Comment (GHWooo) [reply src/components/editor-toolbar/editor-toolbar.tsx]: It's intuitive but can become complex.
I’m curious—how would you handle the situation if a user doesn't want any styles provided at all and prefers to fully customize the styles themselves? 🤔

Comment (rosielsh) [reply src/components/editor-toolbar/editor-toolbar.tsx]: Well, I haven't thought about that. I think it's basically a work that goes beyond the philosophy of the headless UI to process inside the components.

To improve this, I will implement it by separating and distributing css.

Therefore, it will be available as below. ☺️

Example of applying a default style
```javascript
import '@marka-editor/styles/editorToolbar.css';

<Editor.Toolbar>
  {/* toolbar content */}
</Editor.Toolbar>
```

Examples used with custom styles
```javascript
import '@marka-editor/styles/editorToolbar.css';

<Editor.Toolbar style={{ backgroundColor: 'lightgray' }}>
  {/* toolbar content */}
</Editor.Toolbar>
```

I will upload the work of separating styles through additional PR.

Comment (rosielsh) [reply src/components/editor-toolbar/editor-toolbar.tsx]: And I uploaded an additional commit to remove the style. ([Check corresponding commit](https://github.com/rosielsh/marka-editor/pull/28/commits/7ddb8cdbd68be418230055822bd7aa4ce4c0348b))

Comment (GHWooo) [reply src/components/editor-toolbar/editor-toolbar.tsx]: Sounds good. Let's revisit it when we're applying the styles again.

**Review by GHWooo — APPROVED** [review]
LGTM👍
